openlcs/packages/views.py

```diff
@@ -933,12 +934,55 @@ def post(self, request, *args, **kwargs):
                               "source_scan_flag": None})
 
 
+class ComponentFilter(django_filters.FilterSet):
+
+    name = django_filters.CharFilter(field_name='name', lookup_expr='iexact')
+    type = django_filters.CharFilter(
+            field_name='type', lookup_expr='icontains')
+    version = django_filters.CharFilter(
+            field_name='version', lookup_expr='iexact')
+    release = django_filters.CharFilter(
+            field_name='release', lookup_expr='iexact')
+    arch = django_filters.CharFilter(field_name='arch', lookup_expr='iexact')
+    purl = django_filters.CharFilter(
+            field_name='purl', lookup_expr='icontains')
+    uuid = django_filters.CharFilter(field_name='uuid', lookup_expr='iexact')
+    summary_license = django_filters.CharFilter(
+            field_name='summary_license', lookup_expr='iexact')
+    is_source = django_filters.BooleanFilter(field_name='is_source')
+    synced = django_filters.BooleanFilter(field_name='synced')
+    source__name = django_filters.CharFilter(lookup_expr='iexact')
+    source__state = django_filters.NumberFilter()
+    source__archive_type = django_filters.CharFilter(lookup_expr='iexact')
+    source__scan_flag = django_filters.CharFilter(lookup_expr='icontains')
+
+    class Meta:
+        model = Component
+
+        fields = (
+            'name',
+            'type',
+            'version',
+            'release',
+            'arch',
+            'purl',
+            'uuid',
+            'summary_license',
+            'is_source',
+            'synced'
+        )
+
+
 class ComponentViewSet(ModelViewSet, PackageImportTransactionMixin):
     """
     API endpoint that allows components to be viewed.
     """
     queryset = Component.objects.all()
     serializer_class = ComponentSerializer
+    filter_backends = (
+            django_filters.rest_framework.DjangoFilterBackend, SearchFilter)
+    filter_class = ComponentFilter
+    search_fields = ['type', 'name', 'version', 'release', 'arch']
 
     def list(self, request, *args, **kwargs):
         """
```

Comment:
It is better to add some description of how to use the filters here so that users can get them from the restful API page. 
The description of all search fields and how to use them from the command line. such as https://pelc-qe.apps.ocp-c1.prod.psi.redhat.com/rest/v1/tasks/